src/ray/core_worker/core_worker_process.cc

```diff
@@ -21,36 +21,66 @@ namespace ray {
 namespace core {
 namespace {
 
-/// The global instance of `CoreWorkerProcess`.
-std::unique_ptr<CoreWorkerProcess> core_worker_process;
-}  // namespace
+std::unique_ptr<CoreWorkerProcessImpl> core_worker_process;
 
-thread_local std::weak_ptr<CoreWorker> CoreWorkerProcess::current_core_worker_;
+}  // namespace
 
 void CoreWorkerProcess::Initialize(const CoreWorkerOptions &options) {
   RAY_CHECK(!core_worker_process)
       << "The process is already initialized for core worker.";
-  core_worker_process.reset(new CoreWorkerProcess(options));
+  core_worker_process.reset(new CoreWorkerProcessImpl(options));
+
+#ifndef _WIN32
```

Comment:
@kfstorm  this is the only semantical change: we moved `HandleAtExit` registration to `CoreWorkerProcess::Initialize`

Reply:
Is there any particular reason to do this?

Reply:
CoreWorkerProcess is now a static class and doesn't have constructor; also we can put it into CoreWorkerProcessImpl as it doesn't/shouldn't have access to the static class info.

Reply:
I see.